app/models/mentor_membership.rb

```diff
@@ -0,0 +1,25 @@
+# == Schema Information
+#
+# Table name: mentor_memberships
+#
+#  id         :uuid             not null, primary key
+#  type       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  mentor_id  :uuid             not null
+#  school_id  :uuid             not null
+#
+# Indexes
+#
+#  index_mentor_memberships_on_mentor_id  (mentor_id)
+#  index_mentor_memberships_on_school_id  (school_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (mentor_id => mentors.id)
+#  fk_rails_...  (school_id => schools.id)
+#
+class MentorMembership < ApplicationRecord
```

Comment:
Should we be including some sort of database constraint and/or model validation to restrict multiple identical memberships being created?

Right now I can do this:

```ruby
mentor = Mentor.first
school = Placements::School.first
5.times { school.mentors << mentor }
mentor.mentor_memberships.count # => 5
mentor.schools.count { |s| s.id == school.id } # => 5
```

A mentor can be added to the same school multiple times.

Reply:
This will be fixed by https://github.com/DFE-Digital/itt-mentor-services/pull/268#discussion_r1489905897

Reply:
Fixed in c06a141d8659b694ad202fd79569d29eaeceaf2c

Reply:
@CatalinVoineag what do you think about adding a model validation rule, too? So we catch any uniqueness violations in the app/validation layer rather than as an exception from the database.

```ruby
validates_uniqueness_of :mentor_id, scope: [:type, :school_id]
```

Reply:
0dfcc333083a61294db041c68cc2b7f4627cf2cb